src/components/combobox-content.tsx

```diff
@@ -31,24 +32,54 @@ export function ComboboxContent({
     value,
     isMobile,
     isSearching,
+    onKeyboardChange,
 }: ComboboxContentProps) {
     const inputRef = useRef<HTMLInputElement>(null)
+    const [keyboardVisible, setKeyboardVisible] = useState(false)
 
     useEffect(() => {
         if (inputRef.current) {
             inputRef.current.focus()
         }
-    }, [])
+
+        if (typeof window !== 'undefined' && window.visualViewport) {
+            const handleResize = () => {
+                const viewport = window.visualViewport!
+                const heightDiff = window.innerHeight - viewport.height
+                const isKeyboardVisible = heightDiff > 150
+
+                console.log('Viewport metrics:', {
+                    viewportHeight: viewport.height,
+                    windowHeight: window.innerHeight,
+                    heightDiff,
+                    isKeyboardVisible,
+                })
+
+                setKeyboardVisible(isKeyboardVisible)
+                onKeyboardChange?.(isKeyboardVisible ? heightDiff : 0)
+            }
+
+            window.visualViewport.addEventListener('resize', handleResize)
+            handleResize() // Initial check
+            return () => window.visualViewport?.removeEventListener('resize', handleResize)
+        }
+    }, [onKeyboardChange])
 
     return (
-        <Command className={cn(isMobile && 'h-[calc(100vh-10rem)]')}>
+        <Command className={cn('flex h-full flex-col', isMobile && 'h-full')}>
```

Comment:
if you have h-full that applies to every scenario, so isMobile && 'h-full' again will not have any effect and can be removed